src/replication.c

```diff
@@ -1994,23 +1994,32 @@ static void
 handle_update_spec_stats(spec_t *spec, zvol_io_hdr_t *hdr, void *resp)
 {
 	zvol_op_stat_t *stats = (zvol_op_stat_t *)resp;
-	if (hdr->status != ZVOL_OP_STATUS_OK)
+	if ((hdr->status != ZVOL_OP_STATUS_OK) ||
+	    (hdr->len != sizeof (zvol_op_stat_t))) {
```

Comment:
@pawanpraka1 this check is already part of check_header_sanity right?

Reply:
yes